### PR TITLE
Replace unittest with pytest

### PR DESCRIPTION
**Pull request by eivindjahren**
In order to achieve #3798, the tests need to be somewhat independent and standardized. Therefore we need to move as much of the tests as possible to pytest

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: tests/libres_tests/res/enkf/test_enkf.py [hunk 1]
```diff
@@ -38,198 +37,131 @@
...
-    @tmpdir()
-    def test_bootstrap(self):
-        with TestAreaContext("enkf_test", store_area=True) as work_area:
-            work_area.copy_directory(self.case_directory)
-            res_config = ResConfig("simple_config/minimum_config")
-            main = EnKFMain(res_config)
-            self.assertTrue(main, "Load failed")
-
-    @tmpdir()
-    def test_site_condif(self):
```
Comment (mortalisk) [inline tests/libres_tests/res/enkf/test_enkf.py:67]: Need this

File: tests/libres_tests/res/enkf/test_enkf.py [hunk 2]
```diff
@@ -38,198 +37,131 @@
...
-            )
-
-    @tmpdir()
-    def test_site_bootstrap(self):
-        with TestAreaContext("enkf_test", store_area=True):
-            with self.assertRaises(TypeError):
-                EnKFMain(None)
-
-    @tmpdir()
-    def test_default_res_config(self):
```
Comment (mortalisk) [inline tests/libres_tests/res/enkf/test_enkf.py:90]: Need this badly

File: tests/libres_tests/res/enkf/test_enkf_library.py [hunk 3]
```diff
@@ -1,46 +1,27 @@
...
-            with self.assertRaises(NotImplementedError):
-                cls()
+def test_failed_class_creation():
+    classes = [EnkfConfigNode]
 
-    @tmpdir()
-    def test_ecl_config_creation(self):
-        with TestAreaContext("enkf_library_test") as work_area:
-            work_area.copy_directory(self.case_directory)
+    for cls in classes:
```
Comment (mortalisk) [inline tests/libres_tests/res/enkf/test_enkf_library.py:11]: useful loop

**Review by mortalisk — APPROVED** [review]
TLDR: same same, less insane.

File: tests/libres_tests/res/enkf/test_enkf.py (hunk 1 above)
Comment (eivindjahren) [reply tests/libres_tests/res/enkf/test_enkf.py:67]: :rabbit:
